libs/initScripts/computeRSet_with_HCs.js

```diff
@@ -0,0 +1,346 @@
+        params: { limit: NUM_USERS_PER_BATCH_REQUEST, offset: i }
+      })).data.data
+
+      subsetUsers
+      // Filter to users that do not have a CNE
+        .filter(user => !user.creator_node_endpoint)
+        .forEach(user => {
+        // Add userId - wallet mapping
+          userIdToWallet[user.user_id] = user.wallet
+        })
+    } catch (e) {
+      if (e.message.includes('socket hang up')) {
+        await Util.wait(500)
+        console.log('socket hung up.. retrying')
+        i -= NUM_USERS_PER_BATCH_REQUEST
+      } else {
+        console.error(`Could not get users`, e)
+        throw e
+      }
+    }
+  }
+
+  let userIds = Object.keys(userIdToWallet)
+  console.log(`\n${userIds.length} users have no replica sets\nThis is ${userIds.length * 100 / numOfUsers}% of user base`)
+
+  // Get all non-Audius SPs
+  // const audiusInfraSpIdsArr = [1, 2, 3, 4/*, umID*/] // TODO: uncomment later
+
+  // Compute secondaries for users while keeping UM as primary for the new replica set
+  const UMSpId = (await audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceProviderInfoFromEndpoint(USER_METADATA_ENDPOINT)).spID
+
+  const audiusInfraSpIdsArr = [UMSpId]
+  const audiusInfraSpIds = new Set(audiusInfraSpIdsArr) // when UM is registered, exclude it as secondary
+  let spIdToEndpointAndCount = {}
+
+  const sps = await audiusLibs.ethContracts.getServiceProviderList(CONTENT_NODE_TYPE)
+  sps
+    .filter(sp => !audiusInfraSpIds.has(sp.spID))
+    .forEach(sp => {
+      spIdToEndpointAndCount[sp.spID] = { endpoint: sp.endpoint, selected: 0 }
+    })
+
+  console.log('spIds', spIdToEndpointAndCount)
+
+  // Do health check for all non-Audius SPs
+  const healthCheckedSPs = await Promise.all(
+    Object.entries(spIdToEndpointAndCount)
+      .map(entry => performHealthCheck(entry))
+  )
+
+  console.log('Health check responses:', healthCheckedSPs)
+
+  // Filter out unhealthy nodes and retrieve id of each healthy node
+  const spIds = healthCheckedSPs
+    .filter(response => response.status === 200)
+    .map(response => parseInt(response.id))
+
+  console.log('\nComputing replica sets....')
+
+  let userIdToRSet = {}
+  userIds.forEach(id => {
+    // Randomly select two secondaries from spIds
+    let replicaSet = []
+    let secondary1Index = Math.floor(Math.random(0) * spIds.length)
+    let secondary2Index = -1
+    while (secondary2Index === -1 || secondary1Index === secondary2Index) {
+      secondary2Index = Math.floor(Math.random(0) * spIds.length)
+    }
+
+    replicaSet.push(spIds[secondary1Index])
+    replicaSet.push(spIds[secondary2Index])
+
+    userIdToRSet[id] = replicaSet
+
+    // Keep track of number of times the secondary was chosen
+    spIdToEndpointAndCount[spIds[secondary1Index]].selected += 1
+    spIdToEndpointAndCount[spIds[secondary2Index]].selected += 1
+  })
+
+  // Print out user-to-rset mapping
+  console.log('<userIds - replica sets>:\n', userIdToRSet)
```

Comment:
This is part of the output I wanted to see when requesting 'the script output', not just a before/after of a single user

Reply:
even the files attached would be a useful reference

Reply:
will do a new file per batched processing